File: Domain/Trail.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Domain
{
    public class Trail
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Country { get; set; }
        [Range(1, 10)]
        public int Diffculty { get; set; }
        public int? Distance { get; set; }
        public int? RatingId { get; set; }
        public int? PopularityId { get; set; }
        public string ImageTitle { get; set; }  
        public string Description { get; set; }
        
    }
}

File: client/outdoors/src/app/models/trail.ts
export interface Trail {
  id: number;
  name: string;
  country: string;
  diffculty: number;
  distance?: number;
  ratingId?: number;
  popularityId?: number;
  imageTitle: string;
  description?: string;
  recomendedProducts: RecomendedProduct[];
}

export interface RecomendedProduct {
  productId: number;
  productName: string;
  productNameCut: string;
  redirect: string;
}
